src/jit/instr.h

```diff
@@ -276,8 +276,8 @@ enum emitAttr : unsigned
 
 enum InstructionSet
 {
-#ifdef _TARGET_XARCH_
     InstructionSet_ILLEGAL = 0,
```

Comment:
Either  `InstructionSet_NONE` or `InstructionSet_ILLEGAL` should be removed as they are redundant.

Personally I would remove `InstructionSet_ILLEGAL` as it is incorrect.  There is not law against ... 😄

Reply:
Actually, we can make the law .. 😄 

But seriously, convention of making first enum element and last enum element None/Illegal is a kind safeguard which makes coding with enums safer.

Reply:
I certainly have never seen this design pattern nor do I understand why it would be safer.  

If it is intentional I think the comment
```C#
InstructionSet_NONE // No instruction set is available indicating an invalid value
```
should be updated to explain the difference between invalid and illegal.

Reply:
> I certainly have never seen this design pattern nor do I understand why it would be safer.

It is used in multiple places in jit code. instrxarch.h file has a lot of opening and closing invalid values or marks for start and end for given instruction set. Anyway I am not in love with this and if PR will not be accepted that's OK.